src/Language/PureScript/CST/Convert.hs

```diff
@@ -136,11 +136,13 @@ convertType fileName = go
       T.TypeApp ann (Env.tyRecord $> annRec) $ goRow row b
     TypeForall _ kw bindings _ ty -> do
       let
-        mkForAll a b t = do
+        mkForAll a b v t = do
           let ann' = widenLeft (tokAnn $ nameTok a) $ T.getAnnForType t
-          T.ForAll ann' (getIdent $ nameValue a) b t Nothing
-        k (TypeVarKinded (Wrapped _ (Labeled a _ b) _)) = mkForAll a (Just (go b))
-        k (TypeVarName a) = mkForAll a Nothing
+          T.ForAll ann' v (getIdent $ nameValue a) b t Nothing
+        k (TypeVarKinded (Wrapped _ (Labeled (Just _, a) _ b) _)) = mkForAll a (Just (go b)) T.TypeVarVisible
+        k (TypeVarKinded (Wrapped _ (Labeled (_, a) _ b) _)) = mkForAll a (Just (go b)) T.TypeVarInvisible
+        k (TypeVarName (Just _, a)) = mkForAll a Nothing T.TypeVarVisible
+        k (TypeVarName (_, a)) = mkForAll a Nothing T.TypeVarInvisible
```

Comment:
Seems to me that ‘TypeVarKinded’ cases can be combined here and you use ‘maybe’ visibility part. Same for ‘TypeVarName’